src/utils/ValidationHelper.ts

```diff
@@ -102,27 +102,85 @@ export class ValidationHelper {
 		return "";
 	};
 
-	isAddressFormatValid = (jwtPayload: JwtPayload): { errorMessage: string; errorMessageCode: string } => {
-		const personIdentityAddresses: PersonIdentityAddress[] = jwtPayload.shared_claims.address;
-		for (const address of personIdentityAddresses) {
-			if (!this.checkIfValidCountryCode(address.addressCountry)) {
-				return {
-					errorMessage: "Invalid country code: country code is not GB in the postalAddress",
-					errorMessageCode: MessageCodes.INVALID_COUNTRY_CODE,
-				};
-			} else if (!this.checkIfAddressIsValid(address)) {
-				// Validation fails if all the mandatory postalAddress fields- subBuildingName, buildingName, buildingNumber and streetName are missing and is not a valid string or if all the 3 mandatory fields- subBuildingName, buildingName, buildingNumber are missing or not a valid string
-				return {
-					errorMessage: "Missing all or some of mandatory postalAddress fields (subBuildingName, buildingName, buildingNumber and streetName), unable to create the session",
-					errorMessageCode: MessageCodes.MISSING_ALL_MANDATORY_POSTAL_ADDRESS_FIELDS,
-				};
-			}
+	isSharedClaimDataValid(jwtPayload: JwtPayload): { errorMsg: string; errorMsgCode: string } {
+		// Validate user details like givenName, familyName and email
+		if (!this.checkIfValidString([jwtPayload.shared_claims.emailAddress])) {
+			return {
+				errorMsg: "Missing emailAddress from shared claims data",
+				errorMsgCode: MessageCodes.MISSING_PERSON_EMAIL_ADDRESS,
+			};
+		} else if (!jwtPayload.shared_claims.name || !this.isPersonNameValid(jwtPayload.shared_claims.name)) {
+			return {
+				errorMsg: "Missing person's GivenName or FamilyName from shared claims data",
+				errorMsgCode: MessageCodes.MISSING_PERSON_IDENTITY_NAME,
+			};
 		}
 		return {
-			errorMessage: "",
-			errorMessageCode: "",
-		};
-	};
+			errorMsg: "",
+			errorMsgCode: "",
+		};		
+	}
+
+	isPersonNameValid(personName: PersonIdentityName[]) : boolean {		
+		let isValid = true;
+
+		if ( personName.length === 0 ) {
+			isValid = false;
+		} else {
+			for (const name of personName) {
+				const { nameParts } = name;
+				const givenNames: string[] = [];
+				const familyNames: string[] = [];
+				if (nameParts.length === 0 ) {
+					isValid = false;
+				} else {
+					for (const namePart of nameParts) {
+						if (namePart.type === "GivenName" && this.checkIfValidString([namePart.value])) {
+							givenNames.push(namePart.value);
+						}
+						if (namePart.type === "FamilyName" && this.checkIfValidString([namePart.value])) {
```

Comment:
This is interesting because it is actually valid for a user to only have one name, right?

Reply:
Sanka responded: We currently we don’t support single names only. We’ve had a few users be blocked as a result and this has been raised with James N this evening too.

Reply:
Sorry just seeing this, Yoti don't care but yeah the programme doesn't support single names so we need this validation in place to mirror what CIC expects